docs/design/core/presigners/README.md

```diff
@@ -0,0 +1,525 @@
+### Why a different output shape per operation?
+
+This decision is much less obvious. The output shape does not technically need to understand the input shape,
+so the options are: (1) a different, generated output shape per operation, (2) return the `PresignedRequest`
+directly.
+
+The following compares Option 1 and Option 2, in the interest of illustrating why Option 1 was chosen.
+
+**Option 1:** `PresignedGetObjectRequest presignGetObject(GetObjectPresignRequest)`
+
+```Java
+PresignedGetObjectRequest presignedRequest = s3.presignGetObject(...);
+URL presignedUrl = presignedRequest.getUrl();
+```
+
+**Option 2:** `PresignedRequest presignGetObject(GetObjectPresignRequest)``
+
+```Java
+PresignedRequest presignedRequest = s3.presignGetObject(...);
+URL presignedUrl = presignedRequest.getUrl();
+```
+
+**Option 1 Pros:**
+
+1. Makes type-safe execution of presigned requests possible.
+2. Closest to `S3Client` method signature
+
+**Option 2 Pros:**
+
+1. Smaller jar size
+
+**Decision:** Option 1 will be used, because the cost of an empty interface is very small, and it enables
+support for type-safe execution of presigned requests in the future.
+
+*FAQ Below: "What about execution of presigned requests?"*
+
+### What about execution of presigned requests?
+
+The design proposed above makes it possible to execute the signed requests using any HTTP client that implements
+the AWS SDK's HTTP client SPI.
+
+In the future, it would be beneficial to allow a signature user to **execute** a presigned URL using the entire
+SDK stack, instead of just the HTTP client portion. This would enable:
+
+1. Automatic retries, in the case of network or service outages
+2. Response shape unmarshalling, in the case of modeled responses
+3. SDK metric integration (once implemented)
+
+As an example (this is not a design proposal), if the `DynamoDbClient` supported executing a presigned URL, it would be
+beneficial to make sure that the request was for the correct operation, so that the retries and response processing are
+appropriate for the service/operation.
+
+```Java
+DynamoDbClient dynamo = DynamoDbClient.create();
+PresignedPutItemRequest presignedRequest = dynamo.presigner().presignPutItem(...);
+PutItemResponse response = dynamo.putItem(presignedRequest);
+```
+
+### What about non-blocking request presigning?
+
+The proposal above does not distinguish between blocking or non-blocking request presigning. This is because the
+SDK currently only distinguishes between blocking and non-blocking, when it comes to an HTTP client implementation.
+
+The generated presigned request can be executed by a blocking OR non-blocking HTTP client.
+
+In the future, the SDK could implement non-blocking region providers and non-blocking credential providers, at which
+time it could become relevant to distinguish between blocking or non-blocking URL presigners.
+
+For this reason, the `presigner()` method to get to a pre-configured URL presigner will ONLY be included on the
```

Comment:
We have `#utilities` on s3 async client though. Do we want to make presigner consistent with it?

Reply:
Updated the wording to specify that we're not sure and we'll decide when we actually implement the `presigner()` method.